Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -431,7 +431,7 @@ public async Task Submit(SubmitSubmissionServiceModel model)
         await this.submissionsData.Add(newSubmission);
         await this.submissionsData.SaveChanges();
 
-        await this.submissionsForProcessingData.Add(newSubmission.Id);
+        await this.submissionsForProcessingData.Add(newSubmission.Id, this.BuildSubmissionForProcessing(newSubmission, problem, submissionType).ToJson());
```

Comment:
Here is similar - `this.BuildSubmissionForProcessing` is called twice. Can you just call it once?

Reply:
I tried with a variable declared before the transaction is executed, but there was an exception in the database when saving the submission and that is why I am calling this method twice here.

Reply:
This is fixed as discussed.